primitives/src/targeting/eval.rs

```diff
@@ -193,17 +204,34 @@ fn eval(input: &Input, output: &mut Output, rule: &Rule) -> Result<Option<Value>
 
     // basic operators
     let value = match function {
+        Function::MulDiv(first_rule, second_rule, third_rule) => {
+            let product = eval(
+                input,
+                output,
+                &Rule::Function(Function::Mul(first_rule.clone(), second_rule.clone())),
+            )?
+            .ok_or(Error::TypeError)?;
+            let product_rule = Rule::Value(product);
+            let boxed_rule = Box::new(product_rule);
+            let value = eval(
+                input,
+                output,
+                &Rule::Function(Function::Div(boxed_rule, third_rule.clone())),
+            )?
+            .ok_or(Error::TypeError)?;
```

Comment:
loose the `ok_or(...)` and directly return the value. It is already an `Option` and `ok_or` just maps the `None` => `Error::TypeError` in this case, but we don't need it.